x/storage/keeper/keeper.go

```diff
@@ -886,6 +886,10 @@ func (k Keeper) DeleteObject(
 		return types.ErrSourceTypeMismatch
 	}
 
+	if objectInfo.ObjectStatus == types.OBJECT_STATUS_CREATED {
```

Comment:
will introduce a new upgrade in another PR.

Reply:
Yes, this should be a hard fork.